feat(events): add filter rules for prefixEqualsIgnoreCase, suffixEqualsIgnoreCase, wildcard, and anythingBut* matches (#32063)
aws/aws-cdk: packages/aws-cdk-lib/aws-events/README.md

});
```

Matches can also be made case-insensitive, or make use of wildcard matches. For example, to match
object create events for buckets whose name starts with `raw-`, for objects with key matching
the pattern `path/to/object/*.txt` and the requester ends with `.AMAZONAWS.COM`:

```ts
const rule = new events.Rule(this, 'rule', {
  eventPattern: {
    detail: {
      bucket: {
        name: events.Match.prefixEqualsIgnoreCase('raw-'),
      },

      object: {
        key: events.Match.wildcard('path/to/object/*.txt'),
      },

      requester: events.Match.suffixEqualsIgnoreCase('.AMAZONAWS.COM'),
    },
    detailType: events.Match.equalsIgnoreCase('object created'),
  },
});
```

> The "anything but" matchers also support rest parameters. For example:

Reviewer: ```suggestion
The "anything but" matchers allow you to specify multiple arguments. For example:
```
